### PR TITLE
Scripts fixes to prevent teachers from continuously needing to input their PAT

### PR DESCRIPTION
**Pull request by rajbos**
Tested with script 2, 4 and 6

There is also an extra test for the OSTYPE in here, so that the check works on Windows with WSL as well

### CODE REVIEW

**Review by michel-kamp-work — COMMENTED** [review]
this is fine

**Review by michel-kamp-work — COMMENTED** [review]
reviewed

**Review by Radian-n — COMMENTED** [review]
.

**Review by timfry-philips — COMMENTED** [review]
Reviewed

**Comment by n1k17a** [comment]
.

**Review by n1k17a — COMMENTED** [review]
.

**Review by n1k17a — COMMENTED** [review]
.

**Review by ppremk — APPROVED** [review]

**Comment by ppremk** [comment]
merging on behalf @rajbos

File: script/shared_functions [hunk 1]
```diff
@@ -294,8 +294,8 @@ create_pull_request() {
 
   echo -n "Creating pull request: $title... "
 
-  # Wait for 1 second to avoid secondary rate limiting
-  sleep 1
+  # Wait for 2 seconds to avoid secondary rate limiting, 1 second was nog enough
```
Comment (parkerbxyz) [inline script/shared_functions:297]:
```suggestion
  # Wait for 2 seconds to avoid secondary rate limiting (1 second was not enough)
```

**Pull request merged**